docs/docs/porting-to-vial.md

```diff
@@ -14,37 +14,78 @@ redirect_from:
 
 # Porting to Vial
 
-The second part of this tutorial will guide you through porting your keyboard to Vial.
+The second part of this tutorial will guide you through porting your keyboard to Vial. **It assumes you will be adapting source code that was written for and known to work with a recent version of [QMK](https://qmk.fm/).** Keyboard designers are strongly suggested to create a `default` QMK keymap first to familiarize themselves with the firmware creation process and correct any errors.
 
+**If you only have precompiled firmware, you will need to obtain the source code for it before proceeding.** [You have the right to ask the maker of your keyboard for this](https://www.gnu.org/licenses/gpl-faq.en.html#ModifiedJustBinary). Reverse-engineering hardware or software is beyond the scope of this guide.
 
-## 1. Clone the Vial QMK fork
+> Information
+> {: .label .label-green }
+> Compiling Vial firmwware requires some basic knowledge of how to use a command line environment—a bit more than just `qmk compile`. If you are unfamiliar with commands such as `cd` and `pwd`, please read through [this basic primer](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Command_line#basic_built-in_terminal_commands) to learn the commands you'll need to know.
+
+## 1. Prepare Your Build Environment
+
+> Important
+> {: .label .label-red }
+> You will need a QMK environment in order to build firmware for Vial. This is not the same thing as QMK Toolbox. If you do not have this, you must first follow QMK's guide [here](https://docs.qmk.fm/#/newbs_getting_started).
+> Existing QMK environments may need to be updated, but there is no need to uninstall or remove anything. You can simply move to the appropriate folder when compiling firmware for QMK or Vial-QMK.
+
+### Step by step guide:
+1. Clone the latest version of the repository from [https://github.com/vial-kb/vial-qmk](https://github.com/vial-kb/vial-qmk) into a new folder. This folder should be located outside of any existing QMK repository folders. **Nesting `vial-qmk` inside `qmk_firmware` will cause issues, avoid this!**
+   - Install the [GitHub Desktop](https://desktop.github.com) version for your OS. Go to [https://github.com/vial-kb/vial-qmk](https://github.com/vial-kb/vial-qmk) and select "Open in Github Desktop". Follow the guided download/cloning.
+   - **Reference Guide:** Run `git clone https://github.com/vial-kb/vial-qmk`.
```

Comment:
is this two separate options? I don't think we should suggest github desktop for everybody, make sure to indicate it's only for ppl who don't already have git or don't know how to use it

Reply:
I clarified somewhat, but it was inferred that **Reference Guide:** was the advanced guide, and the other was for beginners.